### PR TITLE
- Added BlockSubscriber for subscribing multiple states in single class

### PR DESCRIPTION
**Pull request by jitinsharma**
- Fixed warnings
- Updated docs
- Updated kotlin versions

### CODE REVIEW

File: src/main/kotlin/org/rekotlin/BlockSubscriber.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.rekotlin
+
+/**
+ * BlockSubscriber allows subscription to a single state. This methodology is useful when subscribing multiple states
```
Comment (kmmraj) [inline src/main/kotlin/org/rekotlin/BlockSubscriber.kt]: Please rephrase the comments, 
it should be something like below...
BlockSubscriber allows subscribing to multiple states in a single class.

File: src/main/kotlin/org/rekotlin/BlockSubscriber.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * within single class.
+ *
+ * Ex
+ * val subscriber : BlockSubscriber<MyState>= BlockSubscriber { myState -> }
+ * store.subscribe(subscriber)
+ */
+class BlockSubscriber<S>(private val block: (S) -> Unit) : StoreSubscriber<S> {
```
Comment (kmmraj) [inline src/main/kotlin/org/rekotlin/BlockSubscriber.kt]: Also please update the test cases or add new test cases for testing subscription multiple states in a single class

File: src/main/kotlin/org/rekotlin/Action.kt [hunk 3]
```diff
@@ -28,10 +25,10 @@ package org.rekotlin
  * All actions that want to be able to be dispatched to a store need to conform to this protocol
  * Currently it is just a marker protocol with no requirements.
  */
-interface Action {}
+interface Action
```
Comment (kmmraj) [inline src/main/kotlin/org/rekotlin/Action.kt]: README.md needs to be updated explaining 2 possibilities
1. Use Block subscriber
2. Use delegate class that subscribe to the state, use the delegate class as a member, subscribing and unsubscribing as per the lifecycle needs

File: src/main/kotlin/org/rekotlin/BlockSubscriber.kt (hunk 1 above)
Comment (jitinsharma) [reply src/main/kotlin/org/rekotlin/BlockSubscriber.kt]: Updated

File: src/main/kotlin/org/rekotlin/BlockSubscriber.kt (hunk 2 above)
Comment (jitinsharma) [reply src/main/kotlin/org/rekotlin/BlockSubscriber.kt]: Added test case

File: src/main/kotlin/org/rekotlin/Action.kt (hunk 3 above)
Comment (jitinsharma) [reply src/main/kotlin/org/rekotlin/Action.kt]: Updated README for now. Though i believe one of the samples need to be updated. I'll look into that.

**Pull request merged**